gcc/rust/expand/rust-macro-expand.h

```diff
@@ -90,66 +90,88 @@ class MatchedFragmentContainer
     Repetition,
   };
 
-  MatchedFragmentContainer (std::vector<MatchedFragment> fragments,
-			    Kind kind = Kind::Repetition)
-    : fragments (fragments), kind (kind)
-  {}
+  virtual Kind get_kind () const = 0;
+
+  virtual std::string as_string () const = 0;
 
   /**
    * Create a valid fragment matched zero times. This is useful for repetitions
    * which allow the absence of a fragment, such as * and ?
    */
-  static MatchedFragmentContainer zero ()
-  {
-    return MatchedFragmentContainer ({});
-  }
+  static std::unique_ptr<MatchedFragmentContainer> zero ();
 
   /**
    * Create a valid fragment matched one time
    */
-  static MatchedFragmentContainer metavar (MatchedFragment fragment)
-  {
-    return MatchedFragmentContainer ({fragment}, Kind::MetaVar);
-  }
+  static std::unique_ptr<MatchedFragmentContainer>
+  metavar (MatchedFragment fragment);
 
   /**
    * Add a matched fragment to the container
    */
-  void add_fragment (MatchedFragment fragment)
-  {
-    rust_assert (!is_single_fragment ());
+  void add_fragment (MatchedFragment fragment);
 
-    fragments.emplace_back (fragment);
-  }
+  // const std::string &get_fragment_name () const { return fragment_name; }
```

Comment:
dead code?

Reply:
The diff algorithm produces a strange result here, but that line was moved and not created -- there's a matching ```- // const std::string &get_fragment_name () const { return fragment_name; }``` line in the diff.

Reply:
ah, I see, thanks :) that's fine then, or you can remove it in this PR if you'd like since you're cleaning up the class